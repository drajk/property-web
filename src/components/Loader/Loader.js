import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import styles from './Loader.module.css';
import loaderSvg from './loader.svg';

const Loader = ({ className }) => (
    <img data-testid="loader" className={cx(styles.icon, className)} src={loaderSvg} alt="loader" />
);

Loader.propTypes = {
    className: PropTypes.string,
    iconClassName: PropTypes.string,
};

Loader.defaultProps = {
    className: null,
    iconClassName: null,
};

export default Loader;
