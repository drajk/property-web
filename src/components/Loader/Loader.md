# Loader.js

Shows a loader and allows base class to be overridden.

## Usage

```html
<Loader /> 
<Loader className="fancyClass" />
```
