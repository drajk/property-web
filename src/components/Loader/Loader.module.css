@keyframes rotate {
    100% {
        transform: rotate(360deg);
    }
}

.icon {
    padding: var(--padding);
    text-align: center;
    animation: rotate 0.6s linear infinite;
    height: 100%;
    max-height: 40px;
    max-width: 40px;
    width: 100%;
}
