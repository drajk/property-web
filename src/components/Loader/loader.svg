<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><path d="M24,28.5l-1.3-2.1c3.6-2.3,5.7-6.2,5.7-10.4c0-6.8-5.5-12.4-12.3-12.4c-4.8,0-9.2,2.8-11.2,7.2l-2.3-1 C4.9,4.5,10.2,1.1,16,1.1c8.2,0,14.8,6.7,14.8,14.9C30.8,21.1,28.3,25.8,24,28.5z"></path></svg>