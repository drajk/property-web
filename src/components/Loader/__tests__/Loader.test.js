import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../Loader';

describe('Loader', () => {
    it('renders', () => {
        const tree = renderer.create(<Loader />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders when class is overridden', () => {
        const tree = renderer
            .create(<Loader className="loaderClass" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
