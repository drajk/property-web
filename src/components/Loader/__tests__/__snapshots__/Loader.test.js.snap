// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders 1`] = `
<img
  alt="loader"
  className="icon"
  data-testid="loader"
  src="loader.svg"
/>
`;

exports[`Loader renders when class is overridden 1`] = `
<img
  alt="loader"
  className="icon loaderClass"
  data-testid="loader"
  src="loader.svg"
/>
`;
